tests/spring-cloud-contract-stub-runner-amqp/README.adoc

```diff
@@ -0,0 +1,110 @@
+=== Stub Runner Spring AMQP
+
+Spring Cloud Contract Verifier Stub Runner's messaging module provides an easy way to integrate with Spring AMQP's Rabbit Template.
+For the provided artifacts it will automatically download the stubs and register the required
+routes.
+
+The integration tries to work standalone, that is without interaction with a running RabbitMQ message broker. It expects a `RabbitTemplate` on the application context and uses it as a spring boot test `@SpyBean`.
+Thus it can use the mockito spy functionality to verify and introspect messages sent by the application.
+
+IMPORTANT: In the current status of the implementation tries to find a `MessageListenerAdapter` on the application context to send messages into the application, because we want to avoid interacting with a running bus.
+Using annotation driven listener endpoints (e.g. `@RabbitListener` annotated listener methods) is currently not supported.
+
+==== Adding it to the project
+
+It's enough to have both Spring AMQP and Spring Cloud Contract Stub Runner on classpath.
+Remember to annotate your test class with `@AutoConfigureMessageVerifier`.
+
+==== Examples
+
+===== Stubs structure
+
+Let us assume that we have the following Maven repository with a deployed stubs for the
+`spring-cloud-contract-amqp-test` application.
+
+[source,bash,indent=0]
+----
+└── .m2
+    └── repository
+        └── com
+            └── example
+                └── spring-cloud-contract-amqp-test
+                    ├── 0.4.0-SNAPSHOT
+                    │   ├── spring-cloud-contract-amqp-test-0.4.0-SNAPSHOT.pom
+                    │   ├── spring-cloud-contract-amqp-test-0.4.0-SNAPSHOT-stubs.jar
+                    │   └── maven-metadata-local.xml
+                    └── maven-metadata-local.xml
+----
+
+And the stubs contain the following structure:
+
+[source,bash,indent=0]
+----
+├── META-INF
+│   └── MANIFEST.MF
+└── contracts
+    └── shouldProduceValidPersonData.groovy
+----
+
+Let's consider the following contract:
+
+[source,groovy]
+----
```

Comment:
From what I see this is the last thing to fix. Can you embed this contract in some test class and reference it with `include` directive? That's because if we ever refactor the structure then this part of the doc will get updated too